Medium/1. Shopping Cart Problem/solutions/garimasingh128_Garima Singh/main.cpp

```diff
@@ -0,0 +1,173 @@
+#include <iostream>
+#include <string.h>
+#include <iomanip>
+#include <string.h>
+#include <ctime>
+
+
+using namespace std;
+
+struct items {
+	string itemName;
+	int price;
+	int qty;
+	int dis;
+	int discount;
+	int weight;
+};
+
+
+static struct items object[5];
+string name, ph;
+int pm,n, dis;
+int p;
+	int dist;
+	char address[100];
+float price, total, tax;
+
+
+int main()
+{
+	object[0].itemName = "Basshead earphones";
+	object[0].price = 1200;
+	object[0].discount=1000;
+	object[0].weight=150;
+	object[1].itemName = "Bluetooth computer mouse";
+	object[1].price = 600;
+	object[1].discount=400;
+	object[1].weight=150;
+	object[2].itemName = "Bose earphones";
+	object[2].price = 1000;
+	object[2].discount=900;
+	object[3].weight=150;
+	object[3].itemName = "Bolt earphones";
+	object[3].price = 300;
+	object[3].discount=280;
+	object[3].weight=150;
+	object[4].itemName = "Boat earphones";
+	object[4].price = 1000;
+	object[4].discount=850;
+	object[4].weight=150;
+	for(int i=0;i<5;i++)
+	{
+		object[i].dis=object[i].price-object[i].discount;
+	}
+	
+		cout << "Welcome to GadgetifyWithGSBlr"<<endl;
+	cout<< "Shop address: 311/5 Akshay nagar, Bangalore, Karnataka, India"<<endl;
+	cout<<"Shop contact no: +91 9988776655"<<endl<<endl;
+	cout << "Enter you Name:          ";
+	getline(cin, name);
+	cout << "Enter your phone Number: ";
+	cin >> ph;
+	cout<<"Choose a payment method:"<<endl<<"1. Cash, 2. Card, 3. Online"<<endl<<"enter the corresponding index: ";
+	cin>>pm;
+	cout<<endl;
+	cout<<"Available products on our store:"<<endl;
+	for(int i=0;i<5;i++){
+		cout<<i+1<<".  "<<object[i].itemName<<"   --> Original price:   "<<object[i].price<<"   --> Discounted on the item:  "<<object[i].price-object[i].discount<<"    ---> Final amount to be paid:   "<<object[i].discount<<"   ---> Weight    "<<object[i].weight<<endl;
+	}
+	cout<<endl<<endl;
+
+	cout<<"Enter total no of items you want to buy"<<endl;
+	cin>>n;
+	int selected[n], qty[n];
+	for(int i=0;i<n;i++)
+	{
+	cout <<endl<< "Select the item by its number which you want to buy: "<<endl;
+	cin>>selected[i];
+	cout<<endl;
+	cout << "Quantity of " << object[selected[i]-1].itemName<<endl;
+	cin >> qty[i];
+	dis= dis+ (object[selected[i]-1].dis)*qty[i];
+
+		
+	}
+	
+	
+	cout<<"Select 1 for Takeaway and 2 for Home delivery"<<endl;
+	cin>>p;
+	if(p==2)
+	{
+		
+		cout<<"Enter dist from shop to the delivery address in KM"<<endl;
+		cin>>dist;
+		if(dist > 50)
+	{
+		cout<<"Sorry we don't provide shipping to this address"<<endl;
+	}
+		cout<<"Enter your address"<<endl;
+		scanf(" %[^\n]s", address);
+	}
+
+
+	for(int i=0;i<n;i++)
+	{
+	price = price + qty[i] * object[i].discount;
+	
+	}
+		tax = 0.06 * price;
+	total = tax + price;
+	if(p==2)
+	{
+		if(dist<=5)
+		total=total;
+		if(dist>5 && dist<=20)
+		total=total+30;
+		if(dist>20 && dist<=50)
+		total=total+90;
```

Comment:
For distance from 20kms to 50kms we are charging delivery charges as 60 not 90.

Reply:
I though cumulative amount is to be taken into count!